extensions/amp-mega-menu/0.1/amp-mega-menu.js

```diff
@@ -0,0 +1,363 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AmpEvents} from '../../../src/amp-events';
+import {CSS} from '../../../build/amp-mega-menu-0.1.css';
+import {Keys} from '../../../src/utils/key-codes';
+import {Layout} from '../../../src/layout';
+import {Services} from '../../../src/services';
+import {
+  closestAncestorElementBySelector,
+  isRTL,
+  scopedQuerySelector,
+  scopedQuerySelectorAll,
+  tryFocus,
+} from '../../../src/dom';
+import {dev, userAssert} from '../../../src/log';
+import {isExperimentOn} from '../../../src/experiments';
+import {mod} from '../../../src/utils/math';
+import {setModalAsClosed, setModalAsOpen} from '../../../src/modal';
+import {toArray} from '../../../src/types';
+
+/** @const {string} */
+const TAG = 'amp-mega-menu';
+
+/** @const {string} */
+const ARIA_LABEL_CLOSE = 'Close the mega menu';
```

Comment:
Even though we call it a mega menu, I don't think screenreader users think of it as a mega menu. I would just say "close the menu". In fact, I would consider going one step further and say "Close the ${title} menu".

Reply:
Is `${title}` in this case the menu item headings (e.g. Clothing, Movies, etc.)? If so I'm not sure of the best way to get those values. Perhaps by using `heading.textContent`, but then we'd need to enforce the title text to be directly in the menu button and not one of its children.

Reply:
True. Hmm... tricky. In that case, let's just remove the `mega`, since `mega-menu` is more of a developer construct and not necessarily end user terminology and we can think about adding an option to customise the aria label if there's a strong need for it later.